ПР12

Megrief opened the pull request:
Переработка приложения на почти SingleActivity.

Megrief committed:
Переработка приложения на SingleActivity:
- NavigationGraph
- BottomNavigationMenu
- Заменена верстка MainActivity

Megrief committed:
Переработка приложения на SingleActivity:
- MediaActivity переписана на Fragment
- В MainActivity добавлен NavController, подключен BottomNavigationView.

Megrief committed:
Переработка приложения на SingleActivity:
- SearchActivity преобразована в Fragment
- SettingsActivity преобразована в Fragment
- MediaFragment: Исправлена ошибка, при которой исчезал TabMediator, parentFragmentManager заменен на childFragmentManager

Megrief committed:
Переработка приложения на SingleActivity:
- MainActivity переименована в RootActivity
- Добавлен параметр windowSoftMode в AndroidManifest
- Реализована смена цвета при выделении в BottomNavigationView

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/ui/search/SearchFragment.kt:
Точно savedValue должно быть описано в companion?

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/ui/search/SearchFragment.kt:
```diff
@@ -1,57 +1,68 @@
-class SearchActivity : AppCompatActivity() {
-    private val binding by lazy { ActivitySearchBinding.inflate(LayoutInflater.from(this)) }
+class SearchFragment : Fragment() {
+    private var _binding: FragmentSearchBinding? = null
```
binding у фрагментов стоит обнулять в onDestroyView

DmitrijEgorow commented on app/src/main/res/layout/fragment_search.xml:
На устройствах с увеличенным размером текста немного обрезается

![Screenshot_20230921_110810](https://github.com/Megrief/PlaylistMaker/assets/36837067/8118df8b-de33-4078-b253-ab974badac00)

DmitrijEgorow commented on app/src/main/res/layout/fragment_search.xml:
Название альбома может быть длинным

![Screenshot_20230921_110751](https://github.com/Megrief/PlaylistMaker/assets/36837067/76493610-731b-4418-b8ac-03ddbb23a3f6)

Megrief committed:
Исправлены недочеты в верстке:
- При изменении размера системного шрифта TextView не обрезаются
- AudioplayerActivity trackInfo: при слишком длинном значении верстка не искажается
